integrations/nifi/nifi-rest/nifi-rest-client/nifi-rest-client-api/src/main/java/com/thinkbiganalytics/nifi/rest/client/AbstractNiFiControllerServicesRestClient.java

```diff
@@ -81,6 +95,161 @@ public ControllerServiceDTO updateStateById(@Nonnull final String id, @Nonnull f
+                updateReferences(id, stopComponentsRequest);
+                boolean updatedReferences = ensureComponentsAreOfState(id, "ControllerService", "DISABLED", 5, 500, TimeUnit.MILLISECONDS);
+                if (!updatedReferences) {
+                    //error unable to change the state of the references. ... error
+                    throw new NifiClientRuntimeException(
+                        "Unable to disable other controller service references to this controller service " + controllerService.getName() + " before making the update");
+                }
+            }
+
+        }
+        //mark this controller service as disabled
+        log.info("Disabling the controller service  {} ", controllerService.getName());
+        //update the service and mark it disabled.  this will throw a RuntimeException if it is not successful
+        ControllerServiceDTO updatedService = updateStateByIdWithRetries(controllerService.getId(), "DISABLED", 5, 500, TimeUnit.MILLISECONDS);
+
+        //Perform the update to this controller service
+        log.info("Updating the controller service  {} ", controllerService.getName());
+        updatedService = update(controllerService);
+        //Enable the service
+        updateStateById(controllerService.getId(), NiFiControllerServicesRestClient.State.ENABLED);
+
+        //Enable any controller service references
+        if (!controllerServiceRevisions.isEmpty()) {
+            log.info("Enabling other controller services referencing this controller service  {} ", controllerService.getName());
+            UpdateControllerServiceReferenceRequestEntity enableReferenceServicesRequest = new UpdateControllerServiceReferenceRequestEntity();
+            enableReferenceServicesRequest.setId(controllerService.getId());
+            enableReferenceServicesRequest.setState(NiFiControllerServicesRestClient.State.ENABLED.name());
+            enableReferenceServicesRequest.setReferencingComponentRevisions(controllerServiceRevisions);
+            updateReferences(id, enableReferenceServicesRequest);
+            boolean updatedReferences = ensureComponentsAreOfState(id, "ControllerService", NiFiControllerServicesRestClient.State.ENABLED.name(), 5, 500, TimeUnit.MILLISECONDS);
+            if (!updatedReferences) {
+                //error unable to change the state of the references. ... error
+                throw new NifiClientRuntimeException("The controller service " + controllerService.getName()
+                                                     + " was updated, but it was unable to enable other controller service references to this controller service.  Please visit NiFi to reconcile and fix any controller service issues. "
+                                                     + controllerService.getName());
+            }
+        }
+        if (references.isPresent() && previousProcessorState != null) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2589)